umh-core/pkg/communicator/actions/testutil/memfileinfo.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2025 UMH Systems GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"os"
+	"time"
+)
+
+// MemFileInfo is a simple implementation of os.FileInfo for testing
```

Comment:
Probably the os.FileInfo should expose an interface and the interface could be used in the tests directly for mocking. 
While I agree with the current implementation, I should also point out that the struct MemFileInfo never indicates that it is intended for testing purposes only.